xvec/index.py

```diff
@@ -0,0 +1,262 @@
+        # (by default assume same CRS than the index)
+        label_crs = getattr(label_array, "crs", None)
+        if label_crs is not None and not self._check_crs(label_crs, allow_none=True):
+            self._crs_mismatch_warn(label_crs, stacklevel=4)
+
+        if not np.all(shapely.is_geometry(label_array)):
+            raise ValueError("labels must be shapely.Geometry objects")
+
+        if method == "nearest":
+            indices = self.sindex.nearest(label_array)
+        else:
+            indices = self.sindex.query(label, predicate=method, distance=tolerance)
+
+        # attach dimension names and/or coordinates to positional indexer
+        if isinstance(label, Variable):
+            indices = Variable(label.dims, indices)
+        elif isinstance(label, DataArray):
+            indices = DataArray(indices, coords=label._coords, dims=label.dims)
+
+        return IndexSelResult({self._index.dim: indices})
+
+    def sel(
+        self, labels: dict[Any, Any], method=None, tolerance=None
+    ) -> IndexSelResult:
+        if method is None:
+            return self._index.sel(labels)
+        else:
+            # We reuse here `method` and `tolerance` options of
+            # `xarray.indexes.PandasIndex` as `predicate` and `distance`
+            # options when `labels` is a single geometry.
+            # Xarray currently doesn't support custom options
+            # (see https://github.com/pydata/xarray/issues/7099)
+            return self._sel_sindex(labels, method, tolerance)
+
+    def equals(self, other: Index) -> bool:
+        if not isinstance(other, GeometryIndex):
+            return False
+
+        if not self._check_crs(other.crs, allow_none=True):
+            self._crs_mismatch_warn(other.crs)
+
+        return self._index.equals(other._index)
+
+    def join(
+        self: GeometryIndex, other: GeometryIndex, how: str = "inner"
+    ) -> GeometryIndex:
+        if not self._check_crs(other.crs, allow_none=True):
+            self._crs_mismatch_warn(other.crs)
+
+        index = self._index.join(other._index, how=how)
+        return type(self)(index, self.crs)
+
+    def reindex_like(
+        self, other: GeometryIndex, method=None, tolerance=None
+    ) -> dict[Hashable, Any]:
+        if not self._check_crs(other.crs, allow_none=True):
+            self._crs_mismatch_warn(other.crs)
+
+        return self._index.reindex_like(
+            other._index, method=method, tolerance=tolerance
+        )
+
+    def roll(self, shifts: Mapping[Any, int]) -> GeometryIndex:
+        index = self._index.roll(shifts)
+        return type(self)(index, self.crs)
+
+    def rename(self, name_dict, dims_dict):
+        index = self._index.rename(name_dict, dims_dict)
+        return type(self)(index, self.crs)
+
+    def _repr_inline_(self, max_width: int):
+        srs = _format_crs(self.crs, max_width=max_width)
+        return f"{self.__class__.__name__} (crs={srs})"
```

Comment:
```suggestion
    def _repr_inline_(self, max_width: int):
        if max_width is None:
            max_width = 50
        srs = _format_crs(self.crs, max_width=max_width)
        return f"{self.__class__.__name__} (crs={srs})"
```

Somehow this is raising a TypeError. Cell 6 in the Quickstart notebook:
```py
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
File ~/mambaforge/envs/xvec_dev/lib/python3.10/site-packages/IPython/core/formatters.py:342, in BaseFormatter.__call__(self, obj)
    340     method = get_real_method(obj, self.print_method)
    341     if method is not None:
--> 342         return method()
    343     return None
    344 else:

File ~/mambaforge/envs/xvec_dev/lib/python3.10/site-packages/xarray/core/common.py:167, in AbstractArray._repr_html_(self)
    165 if OPTIONS["display_style"] == "text":
    166     return f"<pre>{escape(repr(self))}</pre>"
--> 167 return formatting_html.array_repr(self)

File ~/mambaforge/envs/xvec_dev/lib/python3.10/site-packages/xarray/core/formatting_html.py:318, in array_repr(arr)
    316 if hasattr(arr, "xindexes"):
    317     indexes = _get_indexes_dict(arr.xindexes)
--> 318     sections.append(index_section(indexes))
    320 sections.append(attr_section(arr.attrs))
    322 return _obj_repr(arr, header_components, sections)

File ~/mambaforge/envs/xvec_dev/lib/python3.10/site-packages/xarray/core/formatting_html.py:195, in _mapping_section(mapping, name, details_func, max_items_collapse, expand_option_name, enabled)
    188 expanded = _get_boolean_with_default(
    189     expand_option_name, n_items < max_items_collapse
    190 )
    191 collapsed = not expanded
    193 return collapsible_section(
    194     name,
--> 195     details=details_func(mapping),
    196     n_items=n_items,
    197     enabled=enabled,
    198     collapsed=collapsed,
    199 )

File ~/mambaforge/envs/xvec_dev/lib/python3.10/site-packages/xarray/core/formatting_html.py:155, in summarize_indexes(indexes)
    154 def summarize_indexes(indexes):
--> 155     indexes_li = "".join(
    156         f"<li class='xr-var-item'>{summarize_index(v, i)}</li>"
    157         for v, i in indexes.items()
    158     )
    159     return f"<ul class='xr-var-list'>{indexes_li}</ul>"

File ~/mambaforge/envs/xvec_dev/lib/python3.10/site-packages/xarray/core/formatting_html.py:156, in <genexpr>(.0)
    154 def summarize_indexes(indexes):
    155     indexes_li = "".join(
--> 156         f"<li class='xr-var-item'>{summarize_index(v, i)}</li>"
    157         for v, i in indexes.items()
    158     )
    159     return f"<ul class='xr-var-list'>{indexes_li}</ul>"

File ~/mambaforge/envs/xvec_dev/lib/python3.10/site-packages/xarray/core/formatting_html.py:139, in summarize_index(coord_names, index)
    136 name = "<br>".join([escape(str(n)) for n in coord_names])
    138 index_id = f"index-{uuid.uuid4()}"
--> 139 preview = escape(inline_index_repr(index))
    140 details = short_index_repr_html(index)
    142 data_icon = _icon("icon-database")

File ~/mambaforge/envs/xvec_dev/lib/python3.10/site-packages/xarray/core/formatting.py:413, in inline_index_repr(index, max_width)
    411 def inline_index_repr(index, max_width=None):
    412     if hasattr(index, "_repr_inline_"):
--> 413         repr_ = index._repr_inline_(max_width=max_width)
    414     else:
    415         # fallback for the `pandas.Index` subclasses from
    416         # `Indexes.get_pandas_indexes` / `xr_obj.indexes`
    417         repr_ = repr(index)

File ~/Git/xvec/xvec/index.py:261, in GeometryIndex._repr_inline_(self, max_width)
    260 def _repr_inline_(self, max_width: int):
--> 261     srs = _format_crs(self.crs, max_width=max_width)
    262     return f"{self.__class__.__name__} (crs={srs})"

File ~/Git/xvec/xvec/index.py:21, in _format_crs(crs, max_width)
     18 else:
     19     srs = "None"
---> 21 return srs if len(srs) <= max_width else " ".join([srs[:max_width], "..."])

TypeError: '<=' not supported between instances of 'int' and 'NoneType'
```

This bit seems to override our default with `None`

```py
File ~/mambaforge/envs/xvec_dev/lib/python3.10/site-packages/xarray/core/formatting.py:413, in inline_index_repr(index, max_width)
    411 def inline_index_repr(index, max_width=None):
    412     if hasattr(index, "_repr_inline_"):
--> 413         repr_ = index._repr_inline_(max_width=max_width)
```
We need  to catch that and override in that case. Either here like in the code suggestion above or in `_format_crs`.

Reply:
Good catch, we can fix it here but I think this should rather be fixed in `xarray.core.formatting.inline_index_repr`? cc @keewis.